src/mongo/MongoGraph.class.php

```diff
@@ -128,8 +128,24 @@ private function add_tarray_to_index($tarray)
         {
             if($key[0] != '_')
             {
-                $predicate = $this->qname_to_uri($key);
-                $predObjects[$predicate] = $this->toGraphValueObject($value);
+                // Make sure the predicate is valid
+                if($this->isValidTripleValue($key)){
```

Comment:
That's true, but it will throw an error rather than an exception if the predicate is an empty string.  I've added ``testAddTripodArrayContainingEmptyPredicate`` to assert that a php error is thrown in this circumstance.  It's not a perfect test, but I think it's probably worth it to not have to unnecessarily validate the predicate.